### PR TITLE
Fixes #21924: Integrate latest build of Metro 2.4.0

### LINKED ISSUE #21924: Update Metro and its dependencies to latest versions
Latest build of Metro 2.4.0  will update following in GF distribution:

saaj-api from 1.3 to 1.4
jaxb-api from 2.2.13 to 2.3.0
mimepull from 1.9.4 to 1.9.7
javax.annotations from 1.3-b01 to its final version (1.3)
jax-ws-api from 2.2.8 to 2.3.0
jaxb, saaj and jaxws RIs to their latest versions
metro itself from 2.3.2 (EE7 version) to latest build of 2.4.0 (EE8 version)

### PR DESCRIPTION
**Pull request by lukasj**
This fixes #21924 by integrating:
saaj-api 1.4
jaxb-api 2.3.0
mimepull 1.9.7
javax.annotations 1.3
jax-ws-api 2.3.0
jaxb-ri 2.3.0-b170624.2321
jaxws-ri 2.3.0-b170625.0012
metro 2.4.0-b170626.1307

### CODE REVIEW

**Comment by lukasj** [comment]
@glassfishrobot Run CI tests please

**Comment by lukasj** [comment]
reviewers: I'm aware that staging repo definition must go away, just wanna get CI to run to see if we can promote artifacts being integrated or there's something to be fixed. Thanks.

**Review by yaminikb — COMMENTED** [review]
Changes look fine except the staging repo snippet

File: appserver/pom.xml [hunk 1]
```diff
@@ -59,6 +59,21 @@
       <url>https://github.com/javaee/glassfish</url>
     </scm>
     
+    <repositories>
+        <!-- TODO: remove me, I'm here for testing only-->
```
Comment (zhengjl) [inline appserver/pom.xml]: should we remove "TODO"?

Comment (lukasj) [inline appserver/pom.xml]: whole repositories block will go away as soon as maven.java.net allows... Integration must be done using 'promoted' or 'releases' repositories which are already defined in parent pom

**Comment by lukasj** [comment]
@glassfishrobot Run CI tests please

File: appserver/tests/appserv-tests/devtests/security/jmac/soap/build.xml [hunk 2]
```diff
@@ -57,7 +57,11 @@
     &testproperties;
     &commonSecurity;
 
-    <target name="all" depends="clean, build-providers, setup, build-deploy, run, undeploy, unsetup"/>
+    <!--<target name="all" depends="clean, build-providers, setup, build-deploy, run, undeploy, unsetup"/>-->
+    <target name="all">
+        <echo message="FIXME: failing test due to a bug in saaj-ri 1.4.0-b11" />
+        <echo message="Known failure see https://jira.oraclecorp.com/jira/browse/IDCINTER-82"/>
```
Comment (pavelbucek) [inline appserver/tests/appserv-tests/devtests/security/jmac/soap/build.xml]: please don't reference internal hosts

Comment (lukasj) [inline appserver/tests/appserv-tests/devtests/security/jmac/soap/build.xml]: I'm not going to fix others

**Comment by yaminikb** [comment]
@glassfishrobot run ci tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**